test/e2e_node/summary_test.go

```diff
@@ -182,21 +182,21 @@ var _ = framework.KubeDescribe("Summary API [NodeConformance]", func() {
 						}),
 						"Memory": ptrMatchAllFields(gstruct.Fields{
 							"Time":            recent(maxStatsAge),
-							"AvailableBytes":  bounded(1*e2evolume.Kb, 80*e2evolume.Mb),
-							"UsageBytes":      bounded(10*e2evolume.Kb, 80*e2evolume.Mb),
+							"AvailableBytes":  gomega.SatisfyAny(gomega.BeNil(), bounded(1*e2evolume.Kb, 80*e2evolume.Mb)),
```

Comment:
These changes required because the code enters in `else`, https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/stats/helper.go#L72 instead of `if`, https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/stats/helper.go#L56

Reply:
What's the path between `cadvisorInfoToCPUandMemoryStats` in https://github.com/kubernetes/kubernetes/blob/e99df0e5a75eb6e86123b56d53e9b7ca0fd00419/pkg/kubelet/stats/helper.go#L36

and the test itself in https://github.com/kubernetes/kubernetes/blob/e99df0e5a75eb6e86123b56d53e9b7ca0fd00419/test/e2e_node/summary_test.go#L41-L43

not seeing any calls to `cadvisorInfoToCPUandMemoryStats` in the test :thinking:

Reply:
`summary_test.go` calls [getNodeSummary](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/util.go#L74) which makes http request to kubelet's `/stats/summary` API endpoint. 

This endpoint is handled by, https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/server/stats/handler.go#L221 which in tern end ups calling `cadvisorInfoToCPUandMemoryStats` via the [CRI stats provider](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/stats/cri_stats_provider.go).

Anyway, while scanning through kubelet logs, I found some more errors that may or may not be related to the fixes below, 

```
Oct 08 09:22:56 localhost kubelet[3337]: E1008 09:22:56.037312    3337 cri_stats_provider.go:377] Failed to get the info of the filesystem with mountpoint "/var/lib/containers/storage/overlay/c4b641869c0da8b82b97bd2f7833f1b2a778acf415c0d3d99a289c1426a95f19/merged": failed to get device for dir "/var/lib/containers/storage/overlay/c4b641869c0da8b82b97bd2f7833f1b2a778acf415c0d3d99a289c1426a95f19/merged": could not find device with major: 0, minor: 71 in cached partitions map.
Oct 08 09:22:56 localhost kubelet[3337]: I1008 09:22:56.042350    3337 cri_stats_provider.go:444] Unable to find network stats for sandbox "01db264e3ae0b4c2e12dbaf72ef2863760e521802ca886b00e0554b66125f754"
```
so I will separate these 2 issues. Let's focus on `UsageNanoCores` in this PR and if needed I can handle the other changes in this PR separately.